llama-index-integrations/vector_stores/llama-index-vector-stores-faiss/llama_index/vector_stores/faiss/map_store.py

```diff
@@ -0,0 +1,188 @@
+    k embeddings, and returns the corresponding indices.
+
+    Args:
+        faiss_index (faiss.IndexIDMap or faiss.IndexIDMap2): Faiss id map index instance
+
+    Examples:
+        `pip install llama-index-vector-stores-faiss faiss-cpu`
+
+        ```python
+        from llama_index.vector_stores.faiss import FaissMapVectorStore
+        import faiss
+
+        # create a faiss index
+        d = 1536  # dimension
+        faiss_index = faiss.IndexFlatL2(d)
+
+        # wrap it in an IDMap or IDMap2
+        id_map_index = faiss.IndexIDMap2(faiss_index)
+
+        vector_store = FaissMapVectorStore(faiss_index=id_map_index)
+        ```
+
+    """
+
+    # _node_id_to_faiss_id_map is used to map the node id to the faiss id
+    _node_id_to_faiss_id_map = PrivateAttr()
+    # _faiss_id_to_node_id_map is used to map the faiss id to the node id
+    _faiss_id_to_node_id_map = PrivateAttr()
+
+    def __init__(
+        self,
+        faiss_index: Any,
+    ) -> None:
+        """Initialize params."""
+        import_err_msg = """
+            `faiss` package not found. For instructions on
+            how to install `faiss` please visit
+            https://github.com/facebookresearch/faiss/wiki/Installing-Faiss
+        """
+        try:
+            import faiss
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        if not isinstance(faiss_index, faiss.IndexIDMap) and not isinstance(faiss_index, faiss.IndexIDMap2):
+            raise ValueError(
+                "FaissVectorMapStore requires a faiss.IndexIDMap or faiss.IndexIDMap2 index. "
+                "Please create an IndexIDMap2 index and pass it to the FaissVectorMapStore."
+            )
+        super().__init__(faiss_index=faiss_index)
+        self._node_id_to_faiss_id_map = {}
+        self._faiss_id_to_node_id_map = {}
+
+    def add(
+        self,
+        nodes: List[BaseNode],
+        **add_kwargs: Any,
+    ) -> List[str]:
+        """
+        Add nodes to index.
+
+        NOTE: in the Faiss vector store, we do not store text in Faiss.
+
+        Args:
+            nodes: List[BaseNode]: list of nodes with embeddings
+
+        """
+        new_ids = []
+        for node in nodes:
+            text_embedding = node.get_embedding()
+            text_embedding_np = np.array(text_embedding, dtype="float32")[np.newaxis, :]
+            self._node_id_to_faiss_id_map[node.id_] = self._faiss_index.ntotal
+            self._faiss_id_to_node_id_map[self._faiss_index.ntotal] = node.id_
+            self._faiss_index.add_with_ids(text_embedding_np, self._faiss_index.ntotal)
+            new_ids.append(node.id_)
+        return new_ids
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        """
+        Delete nodes using with ref_doc_id.
+
+        Args:
+            ref_doc_id (str): The doc_id of the document to delete.
+
+        """
+        # only handle delete on node_ids
```

Comment:
[delete_ref_doc](https://github.com/run-llama/llama_index/blob/e17efe18ae6e21077c79f54d0cb3c7ad3c87182c/llama-index-core/llama_index/core/indices/vector_store/base.py#L377) will call [_delete_from_index_struct](https://github.com/run-llama/llama_index/blob/e17efe18ae6e21077c79f54d0cb3c7ad3c87182c/llama-index-core/llama_index/core/indices/vector_store/base.py#L363C9-L363C34) which will delete the documents by nodes so we dont need to handle any ref_doc_id references.